docs/en/development/developing-api.md

```diff
@@ -0,0 +1,212 @@
+
+When developing a new API, you can refer to the code and documentation of other existing APIs, which will make the development much easier.
+
+**Q: Why should the specification be formulated?**
+
+A: Currently there is a lack of documentation, which makes it difficult for users to use;
+
+The code lacks comments and is not easy for interested contributors to understand, such as : https://github.com/mosn/layotto/issues/112
+
+Adding documents and comments for existing code is a time consuming thing. We hope that the new features developed in the future will have enough docs and comments.
+
+**Q: Following the specification is too much trouble, will it discourage developers who want to contribute code?**
+
+A: **This specification only restricts "what is needed when adding a new Layotto API" (such as a new API to generate distributed self-incrementing id)**, other prs, such as adding a new component or a new SDK, do not need to follow this specification and are free enough.
+
+## Too long; Don't read
+Proposal before development.Make sure the proposal is detailed enough.
+
+Write four documents for users
+- Quick start
+- Usage document
+- API common configuration
+- Component configuration
+
+No need to write design documents, but you need to write detailed comments for proto API and component API and take the comments as doc.
+
+The pr of the new API requires two people to code review.
+
+
+## 1、Publish API proposal to the community and have a full discussion
+### 1.1. Publish a detailed proposal
+#### 1.1.1. Why the proposal should be detailed
+If the proposal is not detailed enough, there is nothing worthy of review by others, and no problem can be found;
+
+The purpose of the review is to brainstorm ideas, and everyone will help analyze the shortcomings of the current design, expose the problems as soon as possible, and avoid rework later.
+
+#### 1.1.2. The content of the proposal
+The proposal needs to include the following:
+
+- demand analysis
+  - Why create this API
+  - Define the boundary of the requirements, which features support and which do not support  
+- Product research on the market
+- grpc/http API design
+- Component API design
+- Explain your design
+
+An example of an excellent proposal：https://github.com/dapr/dapr/issues/2988
+
+### 1.2. Proposal review
+After submitting the proposal, everyone can discuss it in text;
+
+For important or complex API, we can hold community meetings for design review.
+
+
+## 2、Development
+### 2.1. Code specification
+TODO
+
+### 2.2. Test Specification
+- have unit test
+- have client demo, which can be used for demonstration and integration testing
+
+### 2.3. Document specification
+Principle: Documents that need to be written for users; as for design documents for developers, they may expire after a long time and are inconsistent with the code. You don’t need to write them. Explain the design by posting a link to the proposal issue and writing comments in the code.
```

Comment:
These are some grammar mistakes in this sentence

Reply:
I have finished ，please review again